### DISCUSSION TITLE
Justify argument for heading function.

### DISCUSSION THREAD

**Discussion by TiivelS** [typst/typst#4679] [General]
### Description

Possibility to set a boolean for the heading, similar to justify with paragraphs. Should be able to use it hyphenated or not, according to the text hyphenate value. When using justify, text should fill the whole width of the page, when the text is longer than one width (counting the space taken by heading numbering).

### Use Case

Currently, using paragraphs with headings doesn't really work as required, not always filling the whole width of the page. With graphic editors like Word, using justify is really simple and with official documents, often used.

**Comment by Enivex** [comment]
`#show heading: set par(justify: true)`

**Reply by TiivelS** [reply]
Thanks, that's really concise! There is still an issue with typst not hyphenating some words in Estonian, putting them in the next row and therefore the text isn't correctly justified, as in not all rows take up the entire width of the document. Possibly an Estonian language issue with hyphenation. Lorem ipsum text is correctly formatted with justify.
